pkg/hive/clusterimageset_test.go

```diff
@@ -0,0 +1,338 @@
+	}{
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			expectedError:       nil,
+		},
+		{
+			testClusterImageSet: buildInValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			expectedError:       fmt.Errorf("clusterimageset 'name' cannot be empty"),
+		},
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:       nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testClusterImageSet.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testClusterImageSet.Object)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestClusterImageSetExists(t *testing.T) {
+	testCases := []struct {
+		testClusterImageSet *ClusterImageSetBuilder
+		expectedStatus      bool
+	}{
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			expectedStatus:      true,
+		},
+		{
+			testClusterImageSet: buildInValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			expectedStatus:      false,
+		},
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:      false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testClusterImageSet.Exists()
+		assert.Equal(t, exist, testCase.expectedStatus)
+	}
+}
+
+func TestClusterImageSetWithOptions(t *testing.T) {
+	testSettings := buildClusterImageSetClientWithDummyObject()
+	testBuilder := buildValidClusterImageSetBuilder(testSettings).WithOptions(
+		func(builder *ClusterImageSetBuilder) (*ClusterImageSetBuilder, error) {
+			return builder, nil
+		})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+	testBuilder = buildValidClusterImageSetBuilder(testSettings).WithOptions(
+		func(builder *ClusterImageSetBuilder) (*ClusterImageSetBuilder, error) {
+			return builder, fmt.Errorf("error")
+		})
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func TestClusterImageSetWithReleaseImage(t *testing.T) {
+	testCases := []struct {
+		testClusterImageSet *ClusterImageSetBuilder
+		releaseImage        string
+	}{
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			releaseImage:        "release-1-1",
+		},
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(buildClusterImageSetClientWithDummyObject()),
+			releaseImage:        "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testClusterImageSetBuilder := testCase.testClusterImageSet.WithReleaseImage(testCase.releaseImage)
+		if testCase.releaseImage != "" {
+			assert.Equal(t, testCase.releaseImage, testClusterImageSetBuilder.Definition.Spec.ReleaseImage)
+		} else {
+			assert.Equal(t, defaultClusterImageSetRelease, testClusterImageSetBuilder.Definition.Spec.ReleaseImage)
```

Comment:
Not sure I understand this. Shouldn't you have expectedError and check that it is equal to testClusterImageSetBuilder.errorMsg

Reply:
Yes right. Added:
assert.Equal(t, testCase.expectedError, testClusterImageSetBuilder.errorMsg)